Framework/Muon/src/MuonPairingAsymmetry.cpp

```diff
@@ -0,0 +1,476 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidMuon/MuonPairingAsymmetry.h"
+#include "MantidAPI/Algorithm.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidDataObjects/TableWorkspace.h"
+#include "MantidHistogramData/HistogramMath.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/EnabledWhenProperty.h"
+#include "MantidKernel/MandatoryValidator.h"
+#include "MantidMuon/MuonAlgorithmHelper.h"
+
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Kernel;
+
+namespace {
+bool checkPeriodInWorkspaceGroup(const int &period,
+                                 WorkspaceGroup_sptr workspace) {
+  return period <= workspace->getNumberOfEntries();
+}
+
+bool isMultiPeriod(Workspace_sptr ws) {
+  if (auto tmp = boost::dynamic_pointer_cast<WorkspaceGroup>(ws)) {
+    return true;
+  }
+  return false;
+}
+
+int countPeriods(Workspace_sptr ws) {
+  auto tmp = boost::dynamic_pointer_cast<WorkspaceGroup>(ws);
+  return tmp->getNumberOfEntries();
+}
+
+bool checkConsistentPeriods(Workspace_sptr ws1, Workspace_sptr ws2) {
+  if (isMultiPeriod(ws1) && isMultiPeriod(ws2)) {
```

Comment:
unless I am mistaken this line is not needed. I can see 4 cases

1. Multi period and same number -> returns false
2. not multi period and same number -> returns true
3. one multi period and one single -> return true (should this be false?)
4. multi period but different number -> return false